src/reconcile/rec_row.c

```diff
@@ -770,9 +770,11 @@ __wt_rec_row_leaf(
 
         /*
          * If we reconcile an on disk key with a globally visible stop time point and there are no
-         * new updates for that key, skip writing that key.
+         * new updates for that key, skip writing that key. Don't do this additional cleanup work if
+         * a checkpoint is doing the work - we need that to be as streamlined as possible.
          */
-        if (upd == NULL && __wt_txn_tw_stop_visible_all(session, &tw))
+        if (upd == NULL && !F_ISSET(r, WT_REC_CHECKPOINT) &&
+          __wt_txn_tw_stop_visible_all(session, &tw))
```

Comment:
I think I understand this.
By definition, we should never read this OR the associated history store content.
But we're not removing it from the disk image yet (and temporarily wasting space) to not block the checkpoint.

Is that right @kommiharibabu?

Reply:
Yes, your understanding is correct. We are not removing any on-disk tombstones that are globally visible as part of the checkpoint.